compiler/src/dotty/tools/dotc/quoted/PickledQuotes.scala

```diff
@@ -125,6 +125,18 @@ object PickledQuotes {
               val quotedType = evalHole.nn.apply(idx, reifyTypeHoleArgs(args))
               PickledQuotes.quotedTypeToTree(quotedType)
           }
+        case tree @ Template(constr, parents, self, _) if tree.derived.isEmpty =>
+          // Set local dummy as owner when we transform class statements
+          val constr1 = transformSub(constr)
+          val parents1 = transform(tree.parents)
+          val self1 = transformSub(self)
+          val stats1 = flatten(tree.body.mapConserve {
+            case stat: DefTree => transform(stat)
+            case stat =>
+              val localDummy = tree.tpe.termSymbol
+              transform(stat)(using ctx.withOwner(localDummy))
```

Comment:
Could TreeMapWithPreciseStatContexts be reused to avoid a special case?

Reply:
Yes